omod/src/main/java/org/openmrs/module/nigeriaemr/ndrfactory/ClinicalDictionary.java

```diff
@@ -357,6 +357,15 @@ public HIVEncounterType createHIVEncounterType(Patient patient, Date visitDate,
             }
         }
 
+        DateTime dateStoppedRegimen = null;
+        //  nextAppointmentDate = Utils.extractMedicationDuration(visitDate, obsListForOneVisit);
+        obs = Utils.extractObs(Utils.DATE_STOPPED_REGIMEN, obsListForOneVisit);
+        if (obs != null) {
```

Comment:
also check if obs.valueDate is not null to avoid null pointers

Reply:
Hi Wande,

Welldone on fixing this issue. My thought is that when the Doctor selects that RegimenPlan should be "Stop Treatment" for today "VisitDate". The Date Regimen Stopped is the obs.obs_datetime. There is no need to create another concept for Date Regimen Stopped.

Reply:
Hi Bright, 
first Seun fixed the issue, not me so the credit should go to him. Also Seun had a reason for creating the concept and Sylavanus agreed. so Seun kindly state the reason here for Bright.

Reply:
Welldone @Temiye  for fixing this issue, thank you @wandechris for quick feedback. These are just my thought on this issue, A patient has "a state" (Start TX, Stop TX, Continue TX, Restart TX) on every visit (Visit Date or obsdatetime). Instead of creating another concept to hold "date of the states" we just read it from the "encounter_datetime or obs_datetime". @wandechris @wonderfulguru1 @ahmedsylvanus please consider. Doctors get angry when you ask them to enter too many things.

Reply:
> 
> 
Hello Team,

Also put into consideration that there is historical data that has been entered prior before the introduction of this Concept. @wandechris @wonderfulguru1 @ahmedsylvanus @michaelnnakwe 

Thank you

Reply:
ok thank you for your comment
@brightoibe  we will check if that concept doesn't exist then use the encounter date. Sure this will suffice.

Reply:
> also check if obs.valueDate is not null to avoid null pointers

Thank you for this eagle eye. didn't actually think of that. will implement accordingly.